common/persistence/sqlMetadataPersistence.go

```diff
@@ -0,0 +1,383 @@
+				// All integer types are scanned as int64, regardless of the table schema
+				// https://github.com/go-sql-driver/mysql/issues/549
+				Status:      int(result["status"].(int64)),
+				Description: string(result["description"].([]byte)),
+				OwnerEmail:  string(result["owner_email"].([]byte)),
+			},
+			Config: &DomainConfig{
+				Retention:  int32(result["retention"].(int64)),
+				EmitMetric: int2bool("emit_metric"),
+			},
+			ReplicationConfig: &DomainReplicationConfig{
+				ActiveClusterName: GetOrUseDefaultActiveCluster("active",
+					string(result["active_cluster_name"].([]byte))), // TODO TO BE IMPLEMENTED (get rid of "active" placeholder)
+				Clusters: GetOrUseDefaultClusters("active", deserializeClusterConfigs(clusters)), // TODO same
+			},
+			IsGlobalDomain:  int2bool("is_global_domain"),
+			ConfigVersion:   result["config_version"].(int64),
+			FailoverVersion: result["failover_version"].(int64),
+			DBVersion:       result["db_version"].(int64),
+		}, nil
+	}
+
+	// We did not return in the above for-loop because there were no rows.
+	identity := request.Name
+	if len(request.ID) > 0 {
+		identity = request.ID
+	}
+
+	return nil, &workflow.EntityNotExistsError{
+		Message: fmt.Sprintf("Domain %s does not exist.", identity),
+	}
+}
+
+func (m *sqlMetadataManager) UpdateDomain(request *UpdateDomainRequest) error {
+	// Caveat. The Cassandra persistence implementation has the following logic for how to fill in
+	// the "db version" bindings for the update query:
+	// if request.DBVersion <= 0, then (currentVersion, nextVersion) = (nil, 1)
+	// else, (currentVersion, nextVersion) = (request.DBVersion, request.DBVersion + 1)
+
+	clusters, err := gobSerialize(serializeClusterConfigs(request.ReplicationConfig.Clusters))
+	if err != nil {
+		return &workflow.InternalServiceError{
+			Message: fmt.Sprintf("UpdateDomain operation failed. Failed to encode ReplicationConfig.Clusters. Value: %v", request.ReplicationConfig.Clusters),
+		}
+	}
+
+	queryToUse := templateUpdateDomainWhereCurrentVersionIsNullSqlQuery
+	if request.DBVersion > 0 {
+		queryToUse = templateUpdateDomainWhereCurrentVersionIsIntSqlQuery
+	}
+
+	_, err = m.db.NamedExec(queryToUse, &FlatUpdateDomainRequest{
+		DomainInfo:        *(request.Info),
+		DomainConfig:      *(request.Config),
+		ActiveClusterName: request.ReplicationConfig.ActiveClusterName,
+		Clusters:          clusters,
+		ConfigVersion:     request.ConfigVersion,
+		FailoverVersion:   request.FailoverVersion,
+		CurrentDBVersion:  request.DBVersion,
+		NextDBVersion:     request.DBVersion + 1,
+	})
+	if err != nil {
+		return &workflow.InternalServiceError{
+			Message: fmt.Sprintf("UpdateDomain operation failed. Error %v", err),
+		}
+	}
+
+	return nil
+}
+
+// TODO Find a way to get rid of code repetition without using a type switch
+
+func (m *sqlMetadataManager) DeleteDomain(request *DeleteDomainRequest) error {
+	_, err := m.GetDomain(&GetDomainRequest{
```

Comment:
You don't need to do a get before delete in this case. Delete will be noop if there's no row selected. Same for below.